pkg/kubernetes/pvc/v1alpha1/pvc.go

```diff
@@ -0,0 +1,78 @@
+package v1alpha1
+
+import (
+	"strings"
+
+	v1 "k8s.io/api/core/v1"
+)
+
+type predicate func(pvc) bool
+
+type pvc struct {
+	// actual pvc object
+	object *v1.PersistentVolumeClaim
+}
+
+type pvcList struct {
+	// list of pvc objects
+	items []pvc
+}
+
+type listBuilder struct {
+	list    *pvcList
+	filters []predicate
+}
+
+// ListBuilder enables building an instance
+// of pvcList
+func ListBuilder() *listBuilder {
+	return &listBuilder{list: &pvcList{}}
+}
+
+// ContainsName is filter function to filter pvc's
+// based on the name
+func ContainsName(name string) predicate {
+	return func(obj pvc) bool {
+		return strings.Contains(obj.object.GetName(), name)
+	}
+}
+
+// WithListObject builds the list of pvc instances
+// based on the provided pvc instances
+func (b *listBuilder) WithListObject(list *v1.PersistentVolumeClaimList) *listBuilder {
+	if list == nil {
+		return b
+	}
+	for i := range list.Items {
+		b.list.items = append(b.list.items, pvc{&list.Items[i]})
+	}
+	return b
+}
+
+// WithFilter adds filters to filter the pvc list
+func (b *listBuilder) WithFilter(checks ...predicate) *listBuilder {
+	for _, o := range checks {
+		b.filters = append(b.filters, o)
+
+	}
+	return b
+}
+
+// List returns the list of pvc instances that was
+// built by this builder
+func (b *listBuilder) List() (pvcs *v1.PersistentVolumeClaimList) {
+	for _, filter := range b.filters {
+		p := []pvc{}
+		for _, i := range b.list.items {
+			if filter(i) {
+				p = append(p, i)
+			}
+		}
+		b.list.items = p
```

Comment:
Never mutate the original property in final build function.
Here List is the final build function.

Reply:
fixed